tests/integration-tests/tests/schedulers/test_slurm.py

```diff
@@ -38,19 +40,129 @@ def test_slurm(region, pcluster_config_reader, clusters_factory):
     cluster = clusters_factory(cluster_config)
     remote_command_executor = RemoteCommandExecutor(cluster)
 
-    _test_slurm_version(remote_command_executor)
+    # _test_slurm_version(remote_command_executor)
     _test_dynamic_max_cluster_size(remote_command_executor, region, cluster.asg)
     _test_cluster_limits(remote_command_executor, max_queue_size)
     _test_job_dependencies(remote_command_executor, region, cluster.cfn_name, scaledown_idletime, max_queue_size)
     _test_job_arrays_and_parallel_jobs(remote_command_executor, region, cluster.cfn_name, scaledown_idletime)
     assert_overscaling_when_job_submitted_during_scaledown(
         remote_command_executor, "slurm", region, cluster.cfn_name, scaledown_idletime
     )
-    _test_dynamic_dummy_nodes(remote_command_executor, max_queue_size)
+    _test_dynamic_dummy_nodes(remote_command_executor, region, cluster.asg, max_queue_size)
 
     assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
 
 
+@pytest.mark.regions(["us-east-2"])
+@pytest.mark.instances(["g3.8xlarge"])
+@pytest.mark.schedulers(["slurm"])
+@pytest.mark.usefixtures("os", "instance", "scheduler")
+@pytest.mark.slurm_gpu
+def test_slurm_gpu(region, pcluster_config_reader, clusters_factory):
+    """
+    Test Slurm GPU related features.
+
+    Grouped all tests in a single function so that cluster can be reused for all of them.
+    """
+    scaledown_idletime = 3
+    max_queue_size = 3
+    cluster_config = pcluster_config_reader(scaledown_idletime=scaledown_idletime, max_queue_size=max_queue_size)
+    cluster = clusters_factory(cluster_config)
+    remote_command_executor = RemoteCommandExecutor(cluster)
+
+    _gpu_test_scaleup(remote_command_executor, region, cluster.asg, cluster.cfn_name, scaledown_idletime, num_gpus=2)
+    _test_dynamic_dummy_nodes(remote_command_executor, region, cluster.asg, max_queue_size, slots=32, gpus=2)
+    _gpu_test_cluster_limits(remote_command_executor, max_queue_size, 2)
+    _gpu_resource_check(remote_command_executor)
+    _gpu_test_conflicting_options(remote_command_executor, 2)
+
+    assert_no_errors_in_logs(remote_command_executor, ["/var/log/sqswatcher", "/var/log/jobwatcher"])
+
+
+def _gpu_test_cluster_limits(remote_command_executor, max_queue_size, num_gpus):
+    """Test edge cases regarding the number of GPUs."""
+    logging.info("Testing scheduler does not accept jobs when requesting for more GPUs than available")
+
+    # Expect commands below to fail with exit 1
+    result = remote_command_executor.run_remote_command(
```

Comment:
Extend submit_job to take GPU options.

Reply:
I don't think abstraction is very valuable in this case. There are many slurm options that we are only using for GPU tests. We will have to end up either setup a variable in `submit_job` for each of these option or setup an `arbitrary_args` variable. However either way we would ended up almost typing the entire command anyway, and I think that's not as clear as just seeing the command as is.

Reply:
I agree with you for some of the commands, for others I'd lean towards using the scheduler functions because it returns the job id, allows you to check the job status easily, and is centralized logic.